### PR TITLE
889 listening depth

### PR DESCRIPTION
**Pull request by ggarri**
## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://github.com/consensys/orchestrate/blob/master/CHANGELOG.md).

### CODE REVIEW

File: src/tx-listener/tx-listener/use-cases/subscriptions/notify_events.go [hunk 1]
```diff
@@ -35,11 +35,23 @@ func NotifyEventsUseCase(messenger sdk.MessengerAPI, proxyClient sdk.ChainProxyC
 	}
 }
 
-func (uc *notifyEventsUC) Execute(ctx context.Context, sub *entities.Subscription) error {
+func (uc *notifyEventsUC) Execute(ctx context.Context, sub *entities.Subscription, chain *entities.Chain) error {
 	logger := uc.logger.WithField("subscription", sub.UUID)
 
+	var toBlock *big.Int
+
+	if chain.ListenerDepth > 0 {
```
Comment (darioAnongba) [inline src/tx-listener/tx-listener/use-cases/subscriptions/notify_events.go:43]: I don't understand this implementation, isn't the `depth` just the number of blocks to need to wait to consider a receipt or event valid? How is this implementatio working?

File: src/tx-listener/tx-listener/use-cases/jobs/pending.go [hunk 2]
```diff
@@ -36,7 +37,7 @@ func PendingJob(proxyClient sdk.ChainProxyClient,
 	}
 }
 
-func (uc *pendingJobMsg) Execute(ctx context.Context, job *entities.Job) (*entities.Receipt, error) {
+func (uc *pendingJobMsg) Execute(ctx context.Context, job *entities.Job, chain *entities.Chain) (*entities.Receipt, error) {
```
Comment (darioAnongba) [inline src/tx-listener/tx-listener/use-cases/jobs/pending.go:40]: It seems you only need the depth and not the whole chain to be passed :thinking:

File: src/tx-listener/tx-listener/use-cases/subscriptions/notify_events.go (hunk 1 above)
Comment (ggarri) [reply src/tx-listener/tx-listener/use-cases/subscriptions/notify_events.go:43]: Rewrote logic for better clarity. We discard events if the distant to latest block is lowe the chain depth

File: src/tx-listener/tx-listener/use-cases/jobs/pending.go (hunk 2 above)
Comment (ggarri) [reply src/tx-listener/tx-listener/use-cases/jobs/pending.go:40]: indeed, that is an optimization I can do.

**Review by darioAnongba — APPROVED** [review]

**Pull request merged**